src/core/simplify/effectivearea.h

```diff
@@ -91,7 +91,7 @@ struct EFFECTIVE_AREAS
   bool is3d;
   QgsPointSequence inpts;
   areanode *initial_arealist = nullptr;
-  double *res_arealist = nullptr;
+  std::unique_ptr<double[]> res_arealist;
```

Comment:
wouldn't using a std::vector<double> be more natural here ?

Reply:
yes. done